doc/regexp.rdoc

```diff
@@ -1228,6 +1228,34 @@ when regexp.timeout is non-+nil+, that value controls timing out:
   |         nil          |         Float        | Times out in Float seconds. |
   |        Float         |          Any         | Times out in Float seconds. |
 
+== Optimization
+
+Regexp matching time may increase explosively depending on a pattern and an
+target string. A vulnerability caused by this is known as {regular expression
+denial-of-service (ReDoS)}[https://en.wikipedia.org/wiki/ReDoS].
```

Comment:
For certain values of the pattern and target string,
matching time can grow polynomially or exponentially in relation to the input size;
the potential vulnerability arising from this is the {regular expression denial-of-service}[https://en.wikipedia.org/wiki/ReDoS] (ReDoS) attack.

Reply:
Generally, *polynomial* contains linear. However, this is not problematic.